src/channel/mpmc.rs

```diff
@@ -754,15 +775,17 @@ mod if_alloc {
             where
                 T: Send,
             {
-                generic_channel::<parking_lot::RawMutex, T>(capacity)
+                generic_channel::<parking_lot::RawMutex, T, HeapRingBuf<T>>(
+                    capacity,
+                )
             }
 
             /// A [`GenericSender`] implementation backed by [`parking_lot`].
             pub type UnbufferedSender<T> =
-                GenericSender<parking_lot::RawMutex, T>;
+                GenericSender<parking_lot::RawMutex, T, HeapRingBuf<T>>;
```

Comment:
This could now actually be `GenericSender<parking_lot::RawMutex, T, ArrayRingBuf::<T, [T; 0]>>`.

Although I'm not that convinced that the additonal type complexity is worth it, since we already imported `std` and an empty `Vec` doesn't cause an allocation.